lib/ansible/modules/system/selinux.py

```diff
@@ -148,8 +148,8 @@ def set_state(module, state):
 
 
 def set_config_policy(module, policy, configfile):
-    if not os.path.exists('/etc/selinux/%s/policy' % policy):
-        module.fail_json(msg='Policy %s does not exist in /etc/selinux/' % policy)
+    if not os.path.exists(configfile):
```

Comment:
I do not believe this change is correct. The main config file is different than the location of the policy files. The policy directory will exist in `/etc/selinux/[policy name]`.

We would need a separate parameter, `policy_path`, in order to check that the policy exists.

Reply:
On CentOS, `/etc/sysconf/selinux` is a symlink to `/etc/selinux/config`:

```
[root@lab-centos7 ~]# ll /etc/sysconfig/selinux
lrwxrwxrwx. 1 root root 17 Aug 29 06:21 /etc/sysconfig/selinux -> ../selinux/config
```

> On centos there is no such thing as `/etc/selinux/%s/policy`.

This is the path where the policy files for the selected policy, as configured by `SELINUXTYPE` in `/etc/selinux/config`:
```
[root@lab-centos7 ~]# ll /etc/selinux/targeted/
total 16
drwx------. 3 root root  263 Aug 29 06:30 active
-rw-r--r--. 1 root root 2623 Aug 16 17:40 booleans.subs_dist
drwxr-xr-x. 4 root root 4096 Aug 29 06:28 contexts
drwxr-xr-x. 2 root root    6 Aug 16 17:40 logins
drwxr-xr-x. 3 root root   20 Aug 29 06:21 modules
drwxr-xr-x. 2 root root   23 Aug 29 06:30 policy
-rw-------. 1 root root    0 Aug 16 17:10 semanage.read.LOCK
-rw-------. 1 root root    0 Aug 16 17:10 semanage.trans.LOCK
-rw-r--r--. 1 root root  607 Aug 16 17:40 setrans.conf
-rw-r--r--. 1 root root  106 Aug 29 06:30 seusers
```

What the existing code is doing is preventing you from setting a policy in `/etc/selinux/config` that does not have a corresponding set of policy files, which would be an invalid configuration.

If we change that behavior to just checking for the existence of the config file, it would allow someone to configure a policy that does not exist. I don't think that is a good idea.

Reply:
@samdoran Thanks for the clarification. It seems that I got all of this totally wrong since the beginning.

From your last reply, I see what was the point of this line and why it is there. It appears that the "problem" I initially described and attempted to fix, is rather issue with the local environment I was working on.

I will double check on my end and will re-open this if any changes should be made at all.

For now this pull request can be closed.